### PR TITLE
fix: dag order

### PR DESCRIPTION
**Pull request by mfrankovi**
Fix for dag order hash verification. Similar fix is already applied to testnet branch. This is the same fix but with moving the verification inside existing comparePbftBlockScheduleWithDAGblocks_ method

### CODE REVIEW

File: src/consensus/pbft_manager.cpp [hunk 1]
```diff
@@ -1367,18 +1370,76 @@ bool PbftManager::comparePbftBlockScheduleWithDAGblocks_(blk_hash_t const &pbft_
   if (!pbft_block) {
     return false;
   }
-
-  return comparePbftBlockScheduleWithDAGblocks_(*pbft_block).second;
+  return comparePbftBlockScheduleWithDAGblocks_(pbft_block).has_value();
```
Comment (MatusKysel) [inline src/consensus/pbft_manager.cpp:1373]: std::move on block

File: src/consensus/pbft_manager.cpp [hunk 2]
```diff
@@ -1367,18 +1370,76 @@ bool PbftManager::comparePbftBlockScheduleWithDAGblocks_(blk_hash_t const &pbft_
...
+    return std::move(dag_blocks_order);
+  }
+  auto const &anchor_hash = pbft_block->getPivotDagBlockHash();
   auto dag_blocks_order = dag_mgr_->getDagBlockOrder(anchor_hash).second;
   if (!dag_blocks_order.empty()) {
-    return std::make_pair(std::move(dag_blocks_order), true);
+    std::unordered_set<trx_hash_t> trx_set;
+    std::vector<trx_hash_t> transactions_to_query;
+    DbStorage::MultiGetQuery db_query(db_);
+    db_query.append(DbStorage::Columns::dag_blocks, dag_blocks_order);
```
Comment (MatusKysel) [inline src/consensus/pbft_manager.cpp:1391]: isn't is better to get them from dag_blk_mgr_->getDagBlock? I think we can be 99,9% sure those block are still cached at this time

File: src/consensus/pbft_manager.cpp [hunk 3]
```diff
@@ -1367,18 +1370,76 @@ bool PbftManager::comparePbftBlockScheduleWithDAGblocks_(blk_hash_t const &pbft_
...
+    }
+
+    auto calculated_order_hash = calculateOrderHash(dag_blocks_order, non_executed_transactions);
+    if (calculated_order_hash != pbft_block->getOrderHash()) {
+      LOG(log_er_) << "Order hash incorrect. Pbft order hash: " << pbft_block->getOrderHash()
+                   << " . Calculated hash:" << calculated_order_hash << ". Dag order: " << dag_blocks_order
+                   << ". Trx order: " << non_executed_transactions;
+      return {};
+    }
+    cert_sync_block_.pbft_blk = pbft_block;
```
Comment (MatusKysel) [inline src/consensus/pbft_manager.cpp:1437]: std::move

File: src/consensus/pbft_manager.cpp (hunk 2 above)
Comment (mfrankovi) [reply src/consensus/pbft_manager.cpp]: yes, and I added the change for the other two comments

**Review by gaoqi8649 — APPROVED** [review]

**Review by justinsnapp — APPROVED** [review]

**Pull request merged**